librosa/core/constantq.py

```diff
@@ -172,46 +176,32 @@ def cqt(y, sr=22050, hop_length=512, fmin=None, n_bins=84,
 
     # Determine required resampling quality
     Q = float(filter_scale) / (2.0**(1. / bins_per_octave) - 1)
-
     filter_cutoff = fmax_t * (1 + filters.window_bandwidth('hann') / Q)
-
     nyquist = sr / 2.0
-
     if filter_cutoff < audio.BW_FASTEST * nyquist:
         res_type = 'kaiser_fast'
-    elif filter_cutoff < audio.BW_BEST * nyquist:
-        res_type = 'kaiser_best'
     else:
         res_type = 'kaiser_best'
 
-    cqt_resp = []
-
-    len_orig = len(y)
-
     y, sr, hop_length = __early_downsample(y, sr, hop_length,
```

Comment:
AFAICT _early_downsample is a no-op for kaiser_best.  The whole kaiser_fast/kaiser_best logic is making my head hurt.  I think I get it now - if the top octave is close to nyquist, we need to be a little more precise in downsampling.  But the way it all folds together now seems convoluted.  It would be easier to understand if I could quickly tell the "normal case", then see the "kaiser_best" exception as its own block.